### DISCUSSION TITLE
Applying a filter automatically resets expand/collapse default state of other filters

### DISCUSSION THREAD

**Discussion by mbice** [microsoft-search/pnp-modern-search#3030] [Ideas]
**Version used**
4.5.0

**Describe the bug**
When more than one filter group is present and a filter is applied, all the other refiners get reset to their original expand/collapse state. For example, if I have filters set to "expand by default", then I collapse them, they will expand out automatically as soon as any refiner is applied.

**To Reproduce**
1. Configure a search page with 2 or more filters on it
2. Set the filters to "expand by default"
3. Collapse down all filters except one of them
4. Apply a filter
5. Note all other refiners have reverted to their default expanded state (NOTE you can reproduce the bug in reverse as well by setting default state to collapsed and then opening them before applying a filter)

**Expected behavior**
I would expect when a filter is applied that it would respect the pre-existing expand/collapse state of the other filters. In many cases our users collapse down filter sections and having them re-expand automatically is jarring. In V3 it doesn't do this.

**Desktop (please complete the following information):**
 - Latest Chrome

**Comment by FranckyC** [comment]
Hi @mbice, by any chance, is the connection configured in both ways between Search Results and Search Filters?

**Comment by mbice** [comment]
Hi @FranckyC,

I just checked and confirmed the connection is configured both ways between Search Results and Search Filters web parts. Is this an issue?

**Comment by mbice** [comment]
Additionally, I just realized that even just clicking on a filter checkbox will reset the expanded state of the other filters.. I don't even need to apply the filter to reproduce the issue.

**Comment by FranckyC** [comment]
I'll take a look.

**Comment by FranckyC** [comment]
> I cannot repro :/. Can you take a screen record to see what happen (a gif would be ok)?

Nevermind, my mistake (wrong setup). Actually this is the expected behavior since the filter template is re-rendered every time a filter is selected. If the filter hasn't been selected yet, it will keep its default state (see the **vertical.html** file):

```
<pnp-collapsible ... data-default-collapsed="{{#or hasSelectedValues expandByDefault selectedOnce}}false{{else}}true{{/or}}"...
```

The only way to bypass that  would be to use the browser session storage to keep track of (expand/collapse) state.

**Comment by parthibanbalraj** [comment]
@FranckyC, I'm facing the same issue. When I click one tile and when click second one both dropdowns are open until I click title again
